src/node_kernels/SDRSSTTAMSNodeKernel.C

```diff
@@ -0,0 +1,107 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2019 National Renewable Energy Laboratory.                  */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+#include "node_kernels/SDRSSTTAMSNodeKernel.h"
+#include "Realm.h"
+#include "SolutionOptions.h"
+
+#include "stk_mesh/base/MetaData.hpp"
+#include "utils/StkHelpers.h"
+#include <SimdInterface.h>
+
+namespace sierra {
+namespace nalu {
+
```

Comment:
@rcknaus I like the idea of removing duplicate code. Though I am worried about saving off the production field for SST as it would require extra memory and an extra node loop. So I guess there is a tradeoff here between memory and code duplication. Thoughts?

Reply:
Maybe I can keep the compute of production in SST and then just switch to the precomputed average production if using TAMS....

Reply:
Both options would work.  My guess is that the extra nodal loop and field memory would be in the noise since we have a lot of fields and nodal loops executed per time step, but it's nice to check that.